tls/s2n_async_pkey.c

```diff
@@ -409,3 +407,102 @@ S2N_RESULT s2n_async_pkey_sign_free(struct s2n_async_pkey_op *op)
 
     return S2N_RESULT_OK;
 }
+
+int s2n_async_get_op_type(struct s2n_async_pkey_op *op, s2n_async_pkey_op_type * type)
+{
+    POSIX_ENSURE_REF(op);
+    POSIX_ENSURE_REF(type);
+
+    switch (op->type) {
+        case S2N_ASYNC_DECRYPT:
+            *type = S2N_ASYNC_DECRYPT;
+            break;
+        case S2N_ASYNC_SIGN:
+            *type = S2N_ASYNC_SIGN;
+            break;
+        default:
+            POSIX_BAIL(S2N_ERR_NOT_INITIALIZED);
+            break;
+    }
+
+    return S2N_SUCCESS;
+}
+
+int s2n_async_pkey_op_get_input_size(struct s2n_async_pkey_op *op, uint32_t * data_len )
+{
+    POSIX_ENSURE_REF(op);
+    POSIX_ENSURE_REF(data_len);
+
+    if(op->type == S2N_ASYNC_DECRYPT)
+    {
+        struct s2n_async_pkey_decrypt_data *decrypt = &op->op.decrypt;
+        struct s2n_blob *in = &decrypt->encrypted;
+        *data_len = in->size;
+    } else {
+        struct s2n_async_pkey_sign_data *sign = &op->op.sign;
+        struct s2n_hash_state *digest = &sign->digest;
+
+        uint8_t digest_length;
```

Comment:
Nit: For safety / paranoia, init all your variables. Some old systems don't do it for you. So you'll want `uint8_t digest_length = 0`